inferno-core/src/Inferno/Module.hs

```diff
@@ -1,15 +1,24 @@
 {-# LANGUAGE DeriveAnyClass #-}
 {-# LANGUAGE DerivingStrategies #-}
 {-# LANGUAGE NamedFieldPuns #-}
+{-# LANGUAGE TemplateHaskell #-}
 
 module Inferno.Module
   ( Module (..),
     PinnedModule,
+    Prelude (..),
```

Comment:
This file is now basically functions on the `Prelude` type. Should we rename it to `Inferno.Prelude`? But then what do we call `Inferno.Module.Prelude`, which has the actual builtin prelude definition? (I don't think it can be the same file because TH needs the QuasiQuoter defined in a separate module)